lib/swagger/formatter.ex

```diff
@@ -57,71 +38,62 @@ defmodule Xcribe.Swagger.Formatter do
     }
   end
 
-  defp format_body_params(body, request) when is_map(body) do
-    body
-    |> Map.to_list()
-    |> format_params(request)
+  defp include_params(list, params, request, inn) do
+    Enum.reduce(params, list, fn {name, value}, acc ->
+      [build_parameter_object(name, value, request, inn) | acc]
+    end)
   end
 
-  defp format_params([{name, value} | tail], %{controller: controller} = request) do
-    Map.merge(
-      %{
-        name => %{
-          "type" => type_of(value),
-          "description" => Descriptor.get_attr_description(name, controller)
-        }
-      },
-      format_params(tail, request)
-    )
+  defp build_parameter_object(name, value, %{action: action, controller: controller}, inn) do
+    %{
+      "name" => name,
+      "in" => inn,
+      "description" => get_param_description(name, controller, action),
+      "required" => inn == "path",
+      "schema" => %{"type" => type_of(value)}
+    }
   end
 
-  defp format_params([], _), do: %{}
+  defp params_schema_object(params, %{controller: controller}) do
+    Enum.reduce(params, %{}, fn {key, value}, map ->
+      Map.put(map, key, build_simple_schema(key, value, controller))
+    end)
+  end
 
-  defp format_headers([{name, value} | tail]) do
-    Map.merge(
-      %{
-        name => %{"schema" => %{"type" => type_of(value)}}
-      },
-      format_headers(tail)
-    )
+  defp build_simple_schema(name, value, controller) do
+    %{"type" => type_of(value), "description" => get_attr_description(name, controller)}
   end
 
-  defp format_headers([]), do: %{}
+  defp format_headers(headers) do
+    Enum.reduce(headers, %{}, fn {k, v}, acc ->
+      Map.put(acc, k, %{"schema" => %{"type" => type_of(v)}})
+    end)
+  end
 
   defp format_response_body(%{resp_body: body} = request) do
-    %{
-      Descriptor.get_content_type(request) => %{
-        "schema" =>
-          body
-          |> Jason.decode()
-          |> handle_json_decode()
-          |> format_response_body_schema()
-      }
-    }
+    %{get_content_type(request) => %{"schema" => format_body_schema(body)}}
   end
 
-  defp handle_json_decode({:ok, any}), do: any
-  defp handle_json_decode({:error, %{data: data}}), do: data
+  defp format_body_schema(body) do
+    body
+    |> JSON.decode()
+    |> case do
```

Comment:
WDYT of extracting this inline case in a function?

Reply:
Done =D